library/core/src/str/pattern.rs

```diff
@@ -599,6 +602,11 @@ impl Pattern for char {
     {
         self.encode_utf8(&mut [0u8; 4]).strip_suffix_of(haystack)
     }
+
+    #[inline]
+    fn as_utf8_bytes(&self) -> Option<&[u8]> {
+        None
```

Comment:
Why return `None` for `Char`? Doesn't `Char` have a chance to be ascii?

Reply:
> Why return `None` for `Char`? Doesn't `Char` have a chance to be ascii?

Yes ```Char``` and some others are still missing. I just implemented a few to benchmark and to create the draft for some feedback.